src/osd/modules/file/posixsocket.cpp

```diff
@@ -107,6 +107,9 @@ class posix_osd_socket : public osd_file
 
 	virtual error write(void const *buffer, std::uint64_t offset, std::uint32_t count, std::uint32_t &actual) override
 	{
+		if (m_listening) {
+			return error::NONE;
+		}
```

Comment:
I don't think treating a socket as a black hole while it's listening is a good idea.  Also, this makes Win32 and POSIX behaviour diverge.